### PR TITLE
chore: add clang-format rules

### PR DESCRIPTION
**Pull request by aakropotkin**
Adds formatting rules, fixes `Makefile` declaration of `FMT` tool, and formats files.

### CODE REVIEW

**Review by zmitchell — APPROVED** [review]

File: .clang-format [hunk 1]
```diff
@@ -1,225 +1,83 @@
...
-AttributeMacros:
-  - __capability
-BinPackArguments: true
-BinPackParameters: true
-BitFieldColonSpacing: Both
+AllowShortLoopsOnASingleLine: 'true'
+AlwaysBreakAfterDefinitionReturnType: All
+AlwaysBreakAfterReturnType: All
+AlwaysBreakBeforeMultilineStrings: 'false'
+AlwaysBreakTemplateDeclarations: 'Yes'
```
Comment (ysndr) [inline .clang-format:20]:
```suggestion
AlwaysBreakTemplateDeclarations: 'true'
```

This whole config is not quite consistent in its use of quoted and unquoted bools.. Its not a frequented file but looking through it, it's a bit confusing.

File: .clang-format [hunk 2]
```diff
@@ -1,225 +1,83 @@
...
+  AfterStruct:           true
+  AfterUnion:            true
+  BeforeCatch:           true
+  BeforeElse:            true
+  IndentBraces:          true
+  BeforeLambdaBody:      true
+  BeforeWhile:           true
+  SplitEmptyFunction:    false
+  SplitEmptyRecord:      false
+  SplitEmptyNamespace:   false
```
Comment (ysndr) [inline .clang-format:38]:
```suggestion
  AfterClass:            'true'
  AfterControlStatement: 'true'
  AfterEnum:             'false'
  AfterFunction:         'true'
  AfterNamespace:        'false'
  AfterStruct:           'true'
  AfterUnion:            'true'
  BeforeCatch:           'true'
  BeforeElse:            'true'
  IndentBraces:          'true'
  BeforeLambdaBody:      'true'
  BeforeWhile:           'true'
  SplitEmptyFunction:    'false'
  SplitEmptyRecord:      'false'
  SplitEmptyNamespace:   'false'
```

File: Makefile [hunk 3]
```diff
@@ -30,6 +30,7 @@ TEST       ?= test
 DOXYGEN    ?= doxygen
 BEAR       ?= bear
 TIDY       ?= clang-tidy
+FMT        ?= clang-format
```
Comment (ysndr) [inline Makefile:33]: Oh i didn't stage this? oops

**Review by ysndr — COMMENTED** [review]
if we can just get the config file into a consistent situation.

File: .clang-format (hunk 2 above)
Comment (aakropotkin) [reply .clang-format:38]: I unquoted instead

File: Makefile (hunk 3 above)
Comment (aakropotkin) [reply Makefile:33]: No sweat

File: include/flox/flox-flake.hh [hunk 4]
```diff
@@ -34,17 +34,25 @@ namespace flox {
...
+  .applyNixConfig = false /* default */
+  ,
+  .allowUnlocked = true /* default */
+  ,
+  .commitLockFile = false /* default */
+  ,
+  .referenceLockFilePath = std::nullopt /* default */
+  ,
+  .outputLockFilePath = std::nullopt /* default */
+  ,
```
Comment (aakropotkin) [inline include/flox/flox-flake.hh:52]: TODO: this comma thing is hideous and I'll fix it later.

**Pull request merged**